ecommerce/serializers.py

```diff
@@ -365,17 +365,22 @@ def _update_items(cls, basket, items):
         if items:
             # Item updated
             item = items[0]
-            product_id = item.get("product_id")
+            readable_id = item.get("readable_id")
+            readable_id = readable_id.replace(" ", "+")
             run_ids = item.get("run_ids")
-            product = models.Product.objects.get(id=product_id)
+
+            product = models.Product.objects.filter(
+                dj_models.Q(courseruns__courseware_id=readable_id) |
+                dj_models.Q(programs__readable_id=readable_id)
+            ).first()
```

Comment:
@umarmughal824 be sure when user will click on `Enroll Now` button then either there should be the support of `Product_id` too or construct the URL with `readable_id` from that page.
Other wise I am sure, 'No item in basket' will be shown over the checkout page.

Reply:
yes replacing that part with `readable_id`.